doc/contributor/howto-guide-setup-cmake-environment.md

```diff
@@ -0,0 +1,109 @@
+  section from the top-level README.
+* Contributors and developers to `google-cloud-cpp` wanting to work with CMake
+  should read the current document.
+
+The document assumes you are using a Linux workstation running Ubuntu, changing
+the instructions for other distributions or operating systems is left as an
+exercise to the reader. PRs to improve this document are most welcome!
+
+Unless explicitly stated, this document assumes you running these commands at
+the top-level directory of the project, as in:
+
+```console
+cd $HOME
+git clone git@github:<github-username>/google-cloud-cpp.git
+cd google-cloud-cpp
+```
+
+## Install the dependencies in `$HOME/local-cpp`.
+
+In this guide we will install the dependencies in `$HOME/local-cpp`, you can
+pick any directory where you have write permissions. We will then compile the
+project against these dependencies. The installation needs to be done every time
+the version of the dependencies is changed. Once installed you can use them for
+any build or from any other clone.
+
+Configure the super-build to install in `$HOME/local-cpp`. We recommend that
+you use Ninja for this build because it is substantially faster than Make for
+these use-cases. Note that if Ninja is not installed in your workstation you may
+need to remove the `-GNinja` flag:
+
+```console
+cmake -Hsuper -Bcmake-out/si \
+    -DGOOGLE_CLOUD_CPP_EXTERNAL_PREFIX=$HOME/local-cpp -GNinja
+```
+
+Install the dependencies:
+
+```console
+cmake --build cmake-out/si --target project-dependencies -- -j $(nproc)
+```
+
+Now you can use these dependencies multiple times. To use them, add the
+`$HOME/local-cpp` directory to `CMAKE_PREFIX_PATH` when you configure the
+project:
+
+```shell
+cmake -H. -Bcmake-out/home -DCMAKE_PREFIX_PATH=$HOME/local-cpp
+
+# Adjust the number of threads used by modifying parameter for `-j 4`
+cmake --build cmake-out/home -- -j 4
+```
+
+To run the unit tests (only) use:
+
+```shell
+# Verify build by running tests
+(cd cmake-out/home && ctest --output-on-failure -LE integration-test)
+```
+
+If you also want to run the integration tests you need to setup multiple
+[environment variables](/ci/etc/integration-tests-config.sh):
+
+```console
+(cd cmake-out/home && ctest --output-on-failure)
+```
+
+### Changing the Compiler
+
+If your workstation has multiple compilers (or multiple versions of a compiler)
+installed, you can change the compiler using:
+
+```console
+$ CXX=clang++ CC=clang cmake -H. -Bcmake-out/manual
+
+# Then compile and test normally:
+$ cmake --build cmake-out/manual -j 4
+$ (cd cmake-out/manual && ctest --output-on-failure -LE integration-test)
+```
+
+### Changing the Build Type
+
+By default, the system is compiled with optimizations on; if you want to compile
+a debug version, use:
+
+```console
+$ cmake -H. -Bcmake-out/manual -DCMAKE_BUILD_TYPE=Debug
+
+# Then compile and test normally:
+$ cmake --build cmake-out/manual -j 4
```

Comment:
A bit of a nit but it's inconsistent that we use `$(nproc)`, then say `-- -j4`, then further invocations the double dashes are missing. I think the double dashes might even be needed because you're passing `-j 4` to make right?

Reply:
Changed the text to explain when `-j` is needed, and how why `$(nproc)` vs. `-j 4` are useful.

Regarding the `--`, starting with CMake 3.12 the `-j` option is supported directly:

https://cmake.org/cmake/help/v3.12/manual/cmake.1.html#build-a-project

Changed the text around that too.